lib/nrf_modem_lib/nrf91_sockets.c

```diff
@@ -1040,7 +1040,11 @@ static inline int nrf91_socket_offload_poll(struct pollfd *fds, int nfds,
 			fds[i].revents |= POLLNVAL;
 		}
 		if (tmp[i].revents & NRF_POLLHUP) {
-			fds[i].revents |= POLLHUP;
+			/* Apllications are expecting to fail
+			 * in next recv(), thus setting also POLLIN.
+			 * See: https://www.greenend.org.uk/rjk/tech/poll.html
+			 */
+			fds[i].revents |= (POLLHUP | POLLIN);
 		}
```

Comment:
Instead of this change, hopefully we can get the official fix from https://projecttools.nordicsemi.no/jira/browse/NCSDK-9435

Reply:
This is kind of change that would belong in a separate commit (it's not related to the sample).

Reply:
I've removed this change from this PR. It will come with another PR.